test/060-Utils.hpp

```diff
@@ -76,6 +76,14 @@ TEST_CASE( "utilities" ) {
         REQUIRE( flt.size == 1 );
     }
 
+    SECTION( "filter order" ) {
+        REQUIRE( std::is_same_v<refl::trait::get_t<0, decltype(type_list<int, float>{})>, int> );
+        constexpr type_list<int, float> flt = util::filter(type_list<int, float>{}, [](auto ) { return true; });
+        REQUIRE( flt.size == 2 );
+        constexpr int first = find_first(type_list<int, float>{}, [](auto ) { return true; });
+        REQUIRE( std::is_same_v<decltype(first), const int> );
```

Comment:
The funny thing with most of the refl-cpp tests is that they wouldn't even compile if the types were not correct. This is the case here as well (`first` is explicitly typed as `int`). Nevertheless, I use `REQUIRE` everywhere too.

Reply:
Was thinking of `static_assert` as well but since the other test use `REQUIRE` I went down that same route.

Thanks for merging!